src/amo/containers/App.js

```diff
@@ -74,9 +96,76 @@ export class AppBase extends React.Component {
         'userAgent not in state on App load; using navigator.userAgent.');
       setUserAgent(_navigator.userAgent);
     }
+
+    if (authToken) {
+      log.info('Setting a logout timer when the token expires');
+      this.setLogOutTimer(authToken);
+    }
   }
 
-  onViewDesktop = (event, { window_ = window, cookie_ = cookie } = {}) => {
+  componentWillReceiveProps(nextProps: AppProps) {
+    const { authToken } = nextProps;
+    if (authToken) {
+      log.info('Setting a new logout timer when the token expires');
+      this.setLogOutTimer(authToken);
+    }
+  }
+
+  setLogOutTimer(authToken: string) {
+    const { authTokenValidFor, logOutUser } = this.props;
+    const expiresAt = authTokenValidFor;
+    if (!expiresAt) {
+      log.warn(oneLine`configured authTokenValidFor is falsey, not
+        checking for auth expiration`);
+      return;
+    }
+
+    const parts = authToken.split(':');
+    const encodedTimestamp = parts[1];
+    const base62 = getDjangoBase62();
+    let createdAt;
+    try {
+      // This is a UTC Unix timestamp.
+      createdAt = base62.decode(encodedTimestamp);
+    } catch (base62Error) {
+      log.error(
+        `Auth token "${encodedTimestamp}" triggered this base62 error: "${base62Error}"`);
+      return;
+    }
+    // If the encoded timestamp was malformed it will be 0 or negative.
+    if (createdAt <= 0) {
+      log.error(oneLine`Got an invalid timestamp from auth token;
+        encoded value: ${encodedTimestamp}; decoded value: ${createdAt}`);
+      return;
+    }
+    const readableCreatedAt = new Date(createdAt * 1000).toString();
```

Comment:
I didn't think of that. Hmm. Looking at `moment().format()` I don't think it would buy much though.